packages/sdk/src/DeterministicDeployer.ts

```diff
@@ -62,13 +63,21 @@ export class DeterministicDeployer {
     const neededBalance = BigNumber.from(DeterministicDeployer.deploymentGasLimit).mul(DeterministicDeployer.deploymentGasPrice)
     if (bal.lt(neededBalance)) {
       const signer = this.signer ?? this.provider.getSigner()
-      await signer.sendTransaction({
+      const txResponse = await signer.sendTransaction({
         to: DeterministicDeployer.deploymentSignerAddress,
         value: neededBalance,
         gasLimit: DeterministicDeployer.deploymentGasLimit
       })
+      const receipt = await this.provider.waitForTransaction(txResponse.hash, 1, DeterministicDeployer.txWaitTimeout)
+      if (receipt.status !== 1) {
+        throw new Error(`Funding TX failed!\n Receipt: ${JSON.stringify(receipt)} `)
+      }
+    }
+    const txHash = await this.provider.send('eth_sendRawTransaction', [DeterministicDeployer.deploymentTransaction])
```

Comment:
yes, this place need to be fixed. 
already fixed in ethers6 support (by using ethers broadcastTransaction)